pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -232,17 +269,35 @@ func (os *OpenStack) WaitDiskAttached(instanceID string, volumeID string) error
 	}
 
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
-		attached, err := os.diskIsAttached(instanceID, volumeID)
-		if err != nil && !cpoerrors.IsNotFound(err) {
-			// if this is a race condition indicate the volume is deleted
-			// during sleep phase, ignore the error and return attach=false
+		// Check volume attachment status using existing function
+		volumeAttached, err := os.diskIsAttached(instanceID, volumeID)
+		if err != nil {
+			klog.Errorf("Failed to check volume %s attachment status for instance %s: %v", volumeID, instanceID, err)
 			return false, err
 		}
-		return attached, nil
+		if !volumeAttached {
+			klog.V(2).Infof("Volume %s is not yet attached to instance %s", volumeID, instanceID)
+			return false, nil
+		}
+
+		// Check Nova instance attachment status
+		instanceAttached, err := os.isVolumeAttachedToInstance(instanceID, volumeID)
+		if err != nil {
+			klog.Errorf("Failed to check instance %s attachment status for volume %s: %v", instanceID, volumeID, err)
+			return false, err
+		}
+		if !instanceAttached {
+			klog.V(2).Infof("Volume %s is not yet visible in instance %s attached volumes", volumeID, instanceID)
+			return false, nil
```

Comment:
can you propogate the err saying `Volume %s is not yet visible in instance %s attached volumes` ? so that we could see that in pod events and will help us figure out the error rate due to this.